src/components/Article.jsx

```diff
@@ -1,21 +1,40 @@
 import { useState, useEffect } from "react";
-import { getArticle, getComments } from "../api";
-import { useLocation } from "react-router-dom";
+import { getArticle, getComments, updateVotes } from "../api";
+import { useParams } from "react-router-dom";
 import CommentCard from "./CommentCard";
+import ThumbUpOutlinedIcon from "@mui/icons-material/ThumbUpOutlined";
+import ThumbDownOutlinedIcon from "@mui/icons-material/ThumbDownOutlined";
 
 const Article = (props) => {
   const [fullArticle, setFullArticle] = useState({});
   const [comments, setComments] = useState([]);
-  const location = useLocation();
-  const { article_id } = location.state;
+  const [vote, setVote] = useState(0);
+  const [voteError, setVoteError] = useState("");
+  const { article_id } = useParams();
 
   useEffect(() => {
-    getArticle(article_id).then((result) => setFullArticle(result.article));
-  }, []);
+    getArticle(article_id).then((result) => {
+      setFullArticle(result.article);
+      setVote(result.article.votes);
+    });
+  }, [article_id]);
+
```

Comment:
This app would benefit from some loading patterns when content is being loaded from your API. Otherwise the page is blank and the user doesn't know articles are on the way. This is particularly important because the first time your API is woken up it takes a long time. Here are some notes on loading states if you're not sure 🙂
https://notes.northcoders.com/courses/js-front-end/useEffect#loading-patterns